apps/src/aichat/views/TeacherFeedbackFooter.tsx

```diff
@@ -0,0 +1,117 @@
+import classNames from 'classnames';
+import React, {useState} from 'react';
+
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+import Button, {buttonColors} from '@cdo/apps/componentLibrary/button/Button';
+import {EmText} from '@cdo/apps/componentLibrary/typography';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import {AiChatTeacherFeedback as TeacherFeedback} from '@cdo/generated-scripts/sharedConstants';
+
+import aichatI18n from '../locale';
+import {submitTeacherFeedback} from '../redux/aichatRedux';
+import {ChatMessage} from '../types';
+
+import moduleStyles from './teacher-feedback-footer.module.scss';
+
+interface Props {
+  isProfanityViolation: boolean;
+  chatMessage: ChatMessage;
+}
+
+const TeacherFeedbackFooter: React.FC<Props> = ({
+  isProfanityViolation,
+  chatMessage,
+}) => {
+  const dispatch = useAppDispatch();
+  const [thumbsUp, setThumbsUp] = useState(
```

Comment:
I have a feeeeeling that we could do this with a single piece of state (since per the video it seems like the feedback is either thumbsup, thumbsdown, or undefined)? But there's probably a reason this is the way it is :)

Reply:
Yeah good point - I mostly deferred to Cassi given the complexity of the logic here, but will take another look at this while refining UI :)